### PR TITLE
Add graph to calendar page

### PR DESCRIPTION
**Pull request by evulins**
Changes made in this PR:

- Added graph to the Calendar page on both mobile and desktop devices.
- Adjusted all components to match the new designs.

[Trello card](https://trello.com/c/0lfy0FUj/1901-calendar-page-add-graph-to-the-calendar)

### CODE REVIEW

File: app/javascript/react/components/Graph/chartEvents.ts [hunk 1]
```diff
@@ -58,22 +66,53 @@ const addNavigationArrows = (chart: Highcharts.Chart) => {
...
+      iconSize = 46;
+      leftArrowX = -70;
+      rightArrowX = chartWidth + 25;
+      arrowY = chartHeight / 2 - 24;
+    } else if (!isMobile && !isCalendarPage) {
+      iconSize = 48;
+      leftArrowX = 15;
+      rightArrowX = chartWidth - 118;
+      arrowY = chartHeight / 2 - 24;
+    }
```
Comment (Sonje) [inline app/javascript/react/components/Graph/chartEvents.ts:106]: if someone asks me why everything takes so long, I'd point them in this direction - the amount of conditions to place the arrows as in designs can take some time to figure out, implement or fix

File: app/javascript/react/utils/timeFormatter.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import moment from "moment";
+
+export const formattedTime = (time: string, format: string) => {
+  return moment.utc(time).format(format);
```
Comment (Sonje) [inline app/javascript/react/utils/timeFormatter.ts:4]: I think this is the time discrepancy culprit.

**Review by Sonje — APPROVED** [review]
Tremendous work you did here, there are some things that can be slightly improved, but there's no need for that - these changes would only be cosmetic, functionally everything works and looks great.

File: app/javascript/react/utils/timeFormatter.ts (hunk 2 above)
Comment (evulins) [reply app/javascript/react/utils/timeFormatter.ts:4]: This was also my guess  :)

File: app/javascript/react/components/Graph/MeasurementComponent.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  const extremes = useSelector(selectFixedExtremes);
+  const thresholds = useSelector(selectThresholds);
+  const { unitSymbol } = useSelector(selectFixedStreamShortInfo);
+
+  const { minMeasurementValue, maxMeasurementValue, averageValue } = extremes;
+
+  return (
+    <MeasurementContainer>
+      <S.AverageValueContainer>
+        <S.AverageDot $color={getColorForValue(thresholds, averageValue)} />
```
Comment (Sonje) [inline app/javascript/react/components/Graph/MeasurementComponent.tsx:25]: I'd optimize these calculations using memos - by memoizing the result, you avoid recalculating the color values on every render unless the dependencies change 

```
const averageValueColor = useMemo(() => getColorForValue(thresholds, averageValue), [thresholds, averageValue]);
const minMeasurementValueColor = useMemo(() => getColorForValue(thresholds, minMeasurementValue), [thresholds, minMeasurementValue]);
const maxMeasurementValueColor = useMemo(() => getColorForValue(thresholds, maxMeasurementValue), [thresholds, maxMeasurementValue]);
```

**Review by Sonje — APPROVED** [review]
Looks good 👏, one optimization suggested for recalculating colors

**Review by Sonje — APPROVED** [review]
🎉

**Review by JoannaThelema — APPROVED** [review]
I just briefly scrolled through the code. Looks good 🙌

**Review by Sonje — APPROVED** [review]

**Pull request merged**